extensions/resume.md

```diff
@@ -0,0 +1,163 @@
+
+When sent to other clients, the `RESUMED` message MUST have the source of the old client's nickmask. This is illustrated in the examples below.
+
+### Numerics
+
+Here are the new numerics that this extension defines:
+
+| No. | Label                   | Format                                                           |
+| --- | ----------------------- | ---------------------------------------------------------------- |
+| ??? | `ERR_CANNOT_RESUME`     | `:<server> ??? <oldnick> :Cannot resume connection[, <details>]` |
+| ??? | `ERR_HISTORY_TRUNCATED` | `:<server> ??? <client/channel> :Chat history may be truncated`  |
+
+`ERR_CANNOT_RESUME` is used to indicate a failure to resume a connection for a specific reason. The purpose of this numeric is to indicate that resumption was not successful and optionally provide a human-readable explanation. If a more appropriate numeric for conveying the issue already exists (for example, `ERR_NOSUCHNICK` may indicate that the old nickname is no longer being present on the network), then it should be sent after this numeric is sent (i.e. both numerics will be sent).
+
+`ERR_HISTORY_TRUNCATED` is used to indicate that chat history may be truncated for a given client/channel. It may be sent during a `chathistory` batch or otherwise during history playback. Bouncers that reconnect may also find this useful to send to their connected clients, to indicate that there may be missed history.
+
+
+## Connection Registration
+
+Upon connection, clients request the `draft/resume` capability. If they receive both this capability and the SASL capability, they can continue attempting to resume their old connection.
+
+After sending the `NICK` and `USER` commands, clients wishing to resume an old connection send an appropriate `RESUME` command indicating the nickname they wish to take over and how long they've been disconnected for.
+
+After this, the client performs SASL authentication to prove that the indicated old connection does belong to them. If SASL authentication completes successfully, clients are sent the `RESUMED` message and then the regular connection burst (`001`-`005` and all).
+
+Once the connection burst has been received, the server sends the client channel `JOIN` messages as appropriate (and the usual numerics that are sent on joining a channel), and dispatches the relevant messages and numerics to other clients in related channels. The new client is also given the user modes that were active on the old client, as appropriate.
+
+Other clients that have the reconnecting user `MONITOR`'d MUST be sent one `RPL_MONOFFLINE` numeric and one `RPL_MONONLINE` numeric indicating that the user reconnected.
```

Comment:
With `server-time` enabled, what should the timestamp of the `RPL_MONOFFLINE` message be?

Reply:
Added language clarifying that the timestamp on the `RPL_MONOFFLINE` message should be the current time. From the server's perspective the client never really disconnected, so I think it makes sense for it to just send them both as current. The (apparent) time'd out time from the client is given with the `RESUMED` message, so that should suffice.